src/app/api/graphql/announcements/api.announcement.ts

```diff
@@ -1,12 +1,12 @@
 import { API } from "aws-amplify";
 import * as mutations from '../../../../graphql/mutations';
 import * as queries from '../../../../graphql/queries';
-import { GraphQLQuery, graphqlOperation } from '@aws-amplify/api';
-import { CreateAnnouncementsInput, CreateAnnouncementsMutation, DeleteAnnouncementsInput, GetAnnouncementsQuery, ListAnnouncementsQuery, ModelAnnouncementsFilterInput, UpdateAnnouncementsInput, UpdateAnnouncementsMutation } from "../../../../API";
+import { GraphQLQuery, graphqlOperation, GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
+import { AnnouncementsByCoursesIDQuery, AnnouncementsByCoursesIDQueryVariables, CreateAnnouncementsInput, CreateAnnouncementsMutation, DeleteAnnouncementsInput, GetAnnouncementsQuery, ListAnnouncementsQuery, ModelAnnouncementsFilterInput, ModelSortDirection, UpdateAnnouncementsInput, UpdateAnnouncementsMutation } from "../../../../API";
 
 
 export async function createAnnouncement(announcement: CreateAnnouncementsInput) {
-    return await API.graphql<GraphQLQuery<CreateAnnouncementsMutation>>(graphqlOperation(mutations.createAnnouncements, { input: announcement }))
+    return await API.graphql<GraphQLQuery<CreateAnnouncementsMutation>>({...graphqlOperation(mutations.createAnnouncements, { input: announcement }), authMode: GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS})
 }
 
 export async function updateAnnouncement(announcement: UpdateAnnouncementsInput) {
```

Comment:
I don't see the same thing you have done in create mutation. Please check all the updates too.

Reply:
Same for Delete.